3_javascript/javascript1/week3/scripts.js

```diff
@@ -0,0 +1,153 @@
+// HomeWork #3 //
+
+// /// MY FREECODECAMP LINK https://www.freecodecamp.org/SpartanDeBoss
+// /// https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-javascript/build-javascript-objects/
+//  https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-javascript/accessing-object-properties-with-dot-notation/
+//  https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-javascript/accessing-object-properties-with-bracket-notation/
+//  https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-javascript/updating-object-properties/
+//  https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-javascript/add-new-properties-to-a-javascript-object/
+//  https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-javascript/using-objects-for-lookups/
+// // AND THIS MY USERNAME: SpartanDeBoss
+
+//// Item array removal ////
+const names = [
+  'Peter',
+  'Ahmad',
+  'Yana',
+  'kristina',
+  'Rasmus',
+  'Samuel',
+  'katrine',
+  'Tala',
+];
+const nameToRemove = 'Ahmad';
+
+const indexToRemove = names.indexOf(nameToRemove);
+if (indexToRemove !== -1) {
+  names.splice(indexToRemove, 1);
+}
+
+console.log(names);
+
+///// Series duration of my life /////
+const seriesDurations = [
+  {
+    title: 'The chronicles of Narnia',
+    days: 3,
+    hours: 23,
+    minutes: 15,
+  },
+  {
+    title: 'The Wheel Of Time',
+    days: 4,
+    hours: 5,
+    minutes: 11,
+  },
+  {
+    title: 'Banshee',
+    days: 4,
+    hours: 21,
+    minutes: 55,
+  },
+];
+
+function calculatePercentageOfLife(seriesDurations) {
+  const totalMinutesInALife = 80 * 365 * 24 * 60; // Assuming a life expectancy of 80 years
+
+  let totalMinutesSpentWatching = 0;
+
+  for (const series of seriesDurations) {
+    const { days, hours, minutes } = series;
+    const seriesMinutes = days * 24 * 60 + hours * 60 + minutes;
+    totalMinutesSpentWatching += seriesMinutes;
+  }
+
+  const percentageOfLife =
+    (totalMinutesSpentWatching / totalMinutesInALife) * 100;
+  return percentageOfLife.toFixed(3); // Round to 3 decimal places
+}
+
+function logOutSeriesText() {
+  const totalPercentage = calculatePercentageOfLife(seriesDurations);
+  console.log(`In total that is ${totalPercentage}% of my life`);
+}
+
+logOutSeriesText();
+
+///// Save a note /////
+const notes = [];
+
+function saveNote(content, id) {
+  const note = {
+    content: content,
+    id: id,
+  };
+  notes.push(note);
```

Comment:
Good job here. A tip: if the name of the property of the object and the (variable or argument) name are the same, you can do it also in this way.

```javascript
  const note = {
    content,
    id,
  };
```